tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -761,6 +825,54 @@ impl Style {
     }
 }
 
+struct FmtThreadName<'a> {
+    name: &'a str,
+}
+
+impl<'a> FmtThreadName<'a> {
+    pub(crate) fn new(name: &'a str) -> Self {
+        Self { name }
+    }
+}
+
+impl<'a> fmt::Display for FmtThreadName<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use std::sync::atomic::{
+            AtomicUsize,
+            Ordering::{AcqRel, Acquire, Relaxed},
+        };
+
+        static MAX_LEN: AtomicUsize = AtomicUsize::new(0);
+        let len = self.name.len();
+        let mut max_len = MAX_LEN.load(Relaxed);
+
+        while len > max_len {
+            match MAX_LEN.compare_exchange(max_len, len, AcqRel, Acquire) {
+                Ok(_) => break,
+                Err(actual) => max_len = actual,
+            }
+        }
+
+        write!(f, "{:>width$}", self.name, width = max_len)
+    }
+}
+
+struct FmtThreadId {
+    thread_id: std::thread::ThreadId,
+}
+
+impl FmtThreadId {
+    pub(crate) fn new(thread_id: std::thread::ThreadId) -> Self {
+        Self { thread_id }
+    }
+}
+
+impl fmt::Display for FmtThreadId {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{:>2?}", self.thread_id)
```

Comment:
I don't know if this actually works correctly? This looks like it will pad the ID if it is less than 2 characters. However, the ID is always going to be at least ten characters long ( for `ThreadId(`, a single digit, and `)`), and 11 characters if the ID is 2 digits. I think the width needs to be 
```suggestion
        write!(f, "{:>11?}", self.thread_id)
```

Reply:
That was my initial assumption too but when I pad with 11, this is what I get
```
Jul 15 20:30:34.586  INFO tokio-runtime-worker ThreadId(         13) parent_task{subtasks=10}:subtask{number=8}: tokio_spawny_thing: polling subtask number=8
Jul 15 20:30:34.586 DEBUG                 main ThreadId(          1) parent_task{subtasks=10}: tokio_spawny_thing: creating subtask; number=10
Jul 15 20:30:34.586  INFO tokio-runtime-worker ThreadId(         12) parent_task{subtasks=10}:subtask{number=9
```

Looks like the padding is only done to the value inside `ThreadId` so hence I went with `2`

Maybe this has to do with `ThreadId` being a tuple struct

Reply:
Oh, huh, I didn't realize padding tuple structs worked like that, that's weird! Today I learned!

Padding by 2 seems correct to me in this case!